File: 27-remove-element/27-remove-element.cs
public class Solution {
  public int RemoveElement(int[] nums, int val) {
    int fast = 0, slow = 0;
    while (fast < nums.Length) {
        if (nums[fast] != val) {
            nums[slow] = nums[fast];
            slow++;
        }
        fast++;
    }
    return slow;
  }
}

File: 27-remove-element/27-remove-element.scala
object Solution {
  def removeElement(nums: Array[Int], `val`: Int): Int = {
    var fast = 0
    while (fast < nums.length && nums(fast) != `val`) fast += 1
    var slow = fast
    while (fast < nums.length) {
      if (nums(fast) == `val`) fast += 1
      else {
        nums(slow) = nums(fast)
        slow += 1
        fast += 1
      }
    }
    slow
  }
}